mmaction/localization/ssn_utils.py

```diff
@@ -0,0 +1,355 @@
+import fnmatch
+import glob
+import os
+import os.path as osp
+from itertools import groupby
+
+import mmcv
+import numpy as np
+
+from ..core import compute_average_precision_detection, np_softmax
+from . import temporal_iou
+
+
+def load_localize_proposal_file(filename):
+    """Load the proposal file and split it into many parts which contain one
+    video's information separately.
+
+    Args:
+        filename(str): Path to the proposal file.
+
+    Returns:
+        list: List of all videos' information.
+    """
+    lines = list(open(filename))
+
+    # Split the proposal file into many parts which contain one video's
+    # information separately.
+    groups = groupby(lines, lambda x: x.startswith('#'))
+
+    video_infos = [[x.strip() for x in list(g)] for k, g in groups if not k]
+
+    def parse_group(video_info):
+        """Parse the video's information.
+
+        Args:
+            video_info (list): Information of the video.
+
+        Returns:
+            str (video_id): Video's ID.
+            int (num_frames): Number of frames in the video.
+            list (gt_boxes, proposal_boxes): List of the information of bboxes.
```

Comment:
1. no need to indicated returned variable name.
2. should be list[str, int, list[xx], list[xx]]: (video_id, num_frames, gt_boxes, proposal_boxes)

Reply:
https://github.com/open-mmlab/mmcv/blob/master/mmcv/cnn/bricks/plugin.py#L65-L68